app/controllers/consolidations/AssociateUCRController.scala

```diff
@@ -45,26 +46,29 @@ class AssociateUCRController @Inject()(
 
     mucrOptions match {
       case Some(mucrOptions) => Ok(associateUcrPage(associateUcr.fold(form)(form.fill), mucrOptions))
-      case None              => Redirect(controllers.routes.ChoiceController.displayPage())
+      case None              => throw ReturnToStartException
     }
   }
 
   def submit(): Action[AnyContent] = (authenticate andThen getJourney).async { implicit request =>
-    form
-      .bindFromRequest()
-      .fold(
-        formWithErrors => {
-          request.answersAs[AssociateUcrAnswers].mucrOptions match {
-            case Some(mucr) => Future.successful(BadRequest(associateUcrPage(formWithErrors, mucr)))
-            case None       => Future.successful(Redirect(controllers.routes.ChoiceController.displayPage()))
-          }
-        },
-        formData => {
-          val updatedCache = request.answersAs[AssociateUcrAnswers].copy(associateUcr = Some(formData))
-          movementRepository.upsert(Cache(request.pid, updatedCache)).map { _ =>
-            Redirect(consolidationRoutes.AssociateUCRSummaryController.displayPage())
+    val mucrOptions = request.answersAs[AssociateUcrAnswers].mucrOptions
+
+    if(mucrOptions.isEmpty) throw ReturnToStartException
+    else {
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future.successful(BadRequest(associateUcrPage(formWithErrors, mucrOptions.get))),
+          formData => {
+            val updatedCache = request.answersAs[AssociateUcrAnswers].copy(associateUcr = Some(formData))
+            movementRepository.upsert(Cache(request.pid, updatedCache)).map { _ =>
+              Redirect(consolidationRoutes.AssociateUCRSummaryController.displayPage())
+            }
           }
-        }
-      )
+        )
+    }
   }
+
+  private def processForm() = ???
```

Comment:
Is this used ???

Reply:
🤦‍♂ Thanks, let's mi finish this PR before merging, because as I see I can't prepare something without errors. I'm on it

Reply:
Finished